src/smt/proof_logger.h

```diff
@@ -0,0 +1,90 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2024 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Proof logger utility.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__SMT__PROOF_LOGGER_H
+#define CVC5__SMT__PROOF_LOGGER_H
+
+#include "proof/proof_node.h"
+#include "smt/env_obj.h"
+
+namespace cvc5::internal {
+
+/**
+ * The purpose of this class is output proofs for all reasoning the solver
+ * does on-the-fly. It is enabled when proof logging is enabled.
+ * 
+ * This class receives notifications for three things:
+ * (1) When preprocessing has completed, determining the set of input clauses.
+ * (2) When theory lemmas are learned
+ * (3) When a SAT refutation is derived.
+ * 
+ * Dependending on the proof mode, the notifications for the above three things
+ * may be in the form of ProofNode (if proofs are enabled for that component),
+ * or Node (if proofs are disabled for that component).
+ * 
+ * As with dumped proofs, the granularity of the proofs is subject to the
+ * option proof-granularity.
+ */
+class ProofLogger : protected EnvObj
+{
+ public:
+  /** */
+  ProofLogger(Env& env) : EnvObj(env){}
+  ~ProofLogger(){}
+  /**
+   * Called when preprocessing is complete.
+   * @param input The list of input clauses after preprocessing and conversion
+   * to CNF.
```

Comment:
```suggestion
   * @param input The list of input clauses after preprocessing and conversion
   *              to CNF.
```

Reply:
Is this really what we want? Note that clang-format does not normalize to this if the line was 80+ char.

Reply:
Note that I've updated the PR now so that no @param lines span multiple lines.  Nevertheless we should decide on whether we want this style in the future.

Reply:
IMO it makes it easier to read. It makes it hard to read if it is like this:
```
@param foo sdfsdfsdf
asdfsdg asdfdg.